File: api-openbis-java/source/java/ch/ethz/sis/openbis/generic/asapi/v3/dto/typegroup/update/TypeGroupUpdate.java
/*
 *  Copyright ETH 2025 Zürich, Scientific IT Services
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 *
 */

package ch.ethz.sis.openbis.generic.asapi.v3.dto.typegroup.update;

import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.ObjectToString;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.interfaces.IMetaDataUpdateHolder;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.update.FieldUpdateValue;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.update.IObjectUpdate;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.update.IUpdate;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.update.ListUpdateMapValues;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.typegroup.id.ITypeGroupId;
import ch.systemsx.cisd.base.annotation.JsonObject;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;

@JsonObject("as.dto.typegroup.update.TypeGroupUpdate")
public class TypeGroupUpdate implements IUpdate, IObjectUpdate<ITypeGroupId>, IMetaDataUpdateHolder
{

    @JsonProperty
    private ITypeGroupId typeGroupId;

    @JsonProperty
    private FieldUpdateValue<String> name = new FieldUpdateValue<String>();

    @JsonProperty
    private ListUpdateMapValues metaData = new ListUpdateMapValues();

    @Override
    @JsonIgnore
    public ITypeGroupId getObjectId()
    {
        return getTypeGroupId();
    }

    @JsonIgnore
    public ITypeGroupId getTypeGroupId()
    {
        return typeGroupId;
    }

    @JsonIgnore
    public void setTypeGroupId(ITypeGroupId typeGroupId)
    {
        this.typeGroupId = typeGroupId;
    }

    @JsonIgnore
    public FieldUpdateValue<String> getName()
    {
        return name;
    }

    @JsonIgnore
    public void setName(String name)
    {
        this.name.setValue(name);
    }

    @JsonIgnore
    public ListUpdateMapValues getMetaData()
    {
        return metaData;
    }

    @Override
    public String toString()
    {
        return new ObjectToString(this).append("typeGroupId", typeGroupId).toString();
    }
}

File: api-openbis-javascript/src/v3/as/dto/typegroup/update/TypeGroupUpdate.js
define([ "stjs", "as/dto/common/update/FieldUpdateValue", "as/dto/common/update/ListUpdateMapValues" ], function(stjs, FieldUpdateValue, ListUpdateMapValues) {
	var TypeGroupUpdate = function() {
		this.code = new FieldUpdateValue();
		this.managedInternally = new FieldUpdateValue();
		this.metaData = new ListUpdateMapValues();
	};
	stjs.extend(TypeGroupUpdate, null, [], function(constructor, prototype) {
		prototype['@type'] = 'as.dto.typegroup.update.TypeGroupUpdate';
		constructor.serialVersionUID = 1;
		prototype.typeGroupId = null;
		prototype.code = null;
		prototype.managedInternally = null;
		prototype.metaData = null;

		prototype.getObjectId = function() {
			return this.getTypeGroupId();
		};
		prototype.getTypeGroupId = function() {
			return this.typeGroupId;
		};
		prototype.setTypeGroupId = function(typeGroupId) {
			this.typeGroupId = typeGroupId;
		};
		prototype.getCode = function() {
			return this.code;
		};
		prototype.setCode = function(code) {
			this.code.setValue(code);
		};
		prototype.getManagedInternally = function() {
			return this.managedInternally;
		};
		prototype.setManagedInternally = function(managedInternally) {
			this.managedInternally.setValue(managedInternally);
		};
		prototype.getMetaData = function() {
			return this.metaData;
		};
	}, {
		typeGroupId : "ITypeGroupId",
		code : {
			name : "FieldUpdateValue",
			arguments : [ "String" ]
		},
		managedInternally : {
			name : "FieldUpdateValue",
			arguments : [ "Boolean" ]
		},
		metaData : 'ListUpdateMapValues'
	});
	return TypeGroupUpdate;
})